openstack/auth_env.go

```diff
@@ -22,6 +39,16 @@ func AuthOptionsFromEnv() (gophercloud.AuthOptions, error) {
 	domainID := os.Getenv("OS_DOMAIN_ID")
 	domainName := os.Getenv("OS_DOMAIN_NAME")
 
+	// If OS_PROJECT_ID is set, overwrite tenantID with the value.
+	if v := os.Getenv("OS_PROJECT_ID"); v != "" {
```

Comment:
If Gophercloud had full support for `ProjectName` and `ProjectID` in `AuthOptions`, I could see throwing an error if both Tenant and Project were used. This kind of error check could be added in the future.

Terraform has been doing this kind of permissive setting for months and there hasn't been an issue raised about it, so I think it's pretty safe:

https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/provider.go#L44-L61